packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -53,16 +82,24 @@ async function loadChainId(baseUrl: string): Promise<ChainId> {
 }
 
 export class EthereumConnection implements BcpConnection {
-  public static async establish(baseUrl: string): Promise<EthereumConnection> {
+  public static async establish(baseUrl: string, wsUrl: string | undefined): Promise<EthereumConnection> {
     const chainId = await loadChainId(baseUrl);
-    return new EthereumConnection(baseUrl, chainId);
+    let socketStream;
+    if (wsUrl) {
+      socketStream = new StreamingSocket(wsUrl);
+      socketStream.connect();
+      await socketStream.connected;
+    }
+    return new EthereumConnection(baseUrl, chainId, socketStream);
   }
 
   private readonly baseUrl: string;
   private readonly myChainId: ChainId;
+  private readonly socket: StreamingSocket | undefined;
 
-  constructor(baseUrl: string, chainId: ChainId) {
+  constructor(baseUrl: string, chainId: ChainId, socketStream: StreamingSocket | undefined) {
```

Comment:
`constructor(baseUrl: string, chainId: ChainId, wsUrl: string | undefined)` and connect in here